### DISCUSSION TITLE
difficulties to download

### DISCUSSION THREAD

**Discussion by Maranhao6456** [semiautomaticgit/SemiAutomaticClassificationPlugin#216] [General]
I am not able to download S2 images. When I run run and choose the folder, a 4-beep sound comes in and nothing happens.

![scp](https://user-images.githubusercontent.com/67442712/164514800-ca627091-8a77-427a-992d-614829b3b017.jpg)

**Comment by semiautomaticgit** [comment]
Hello, you should first add a preview to the map using the specific button, or uncheck the option "Only if preview in layers"

**Comment by Maranhao6456** [comment]
Hi,  Just as you have the option in the SCP to download L8 and S2 images, you could also download CBERS 4 and 4A images. Have you ever thought about that possibility?

**Reply by semiautomaticgit** [reply]
Thank you for this information.
I'll think about adding these sources.

**Comment by Maranhao6456** [comment]
Hi,
With the recent change in access to Sentinel 2 images now via the Copernicus dataspace, what are the instructions for downloading images in SCP?

**Reply by semiautomaticgit** [reply]
Hi @Maranhao6456 , the instructions are quite the same.
You set the search area and range of dates of acquisition to get available images, then you can download the selected images.
Please follow this tutorial https://semiautomaticclassificationmanual.readthedocs.io/en/latest/tutorial_3.html

**Comment by Maranhao6456** [comment]
Hi Luca!

Thanks for the clarifications. I had a question... what do the L2A and RT_L2A files mean?

**Comment by semiautomaticgit** [comment]
You are welcome!

> Thanks for the clarifications. I had a question... what do the L2A and RT_L2A files mean?

L2A are the original files downloaded, while RT_L2A are the same files but converted to decimal values using the scale factor.

**Reply by Maranhao6456** [reply]
Hi,

grateful!

Is there any documentation about these files?

**Reply by semiautomaticgit** [reply]
You are welcome! 

> Is there any documentation about these files?

There is the official documentation at ESA website such as https://sentinel.esa.int/documents/247904/1848117/Sentinel-2-Level-2A-Product-Definition-Document.pdf

**Reply by Maranhao6456** [reply]
Hi Luca,

In SCP, the image recommended for download is HLS (site: earthdata), which according to the available documentation has 30 m resolution (S30 and L30). But when we download through SCP what appears are 2 types of images
at L2A and RT_L2A, with 10m resolution. Any explanation for this difference?

**Reply by semiautomaticgit** [reply]
Hi @Maranhao6456 , the HLS is based on Sentinel-2 data but harmonized with Landsat data  (please see https://hls.gsfc.nasa.gov/), and bands have 30m resolution, useful for time series analysis. 
The original Copernicus Sentinel-2 are available with the corresponding option "Sentinel-2". You can have a look at this tutorial https://semiautomaticclassificationmanual.readthedocs.io/en/latest/tutorial_3.html

**Comment by Maranhao6456** [comment]
Hi Luca,

I am unable to import the signature file. I installed the dependencies but the error still persists. Attached is the error message and also the file I'm trying to import.

If you can help I would appreciate it.

2024-02-12T06:10:23     WARNING    Traceback (most recent call last):
              File "C:\Users/Marcelo Maranhao/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\SemiAutomaticClassificationPlugin\interface\scp_dock.py", line 2430, in import_library_file
              cfg.scp_training.save_temporary_signature_catalog()
             AttributeError: 'NoneType' object has no attribute 'save_temporary_signature_catalog'


[Amostra_Ilha_do_Governador_Cassio.zip](https://github.com/semiautomaticgit/SemiAutomaticClassificationPlugin/files/14236890/Amostra_Ilha_do_Governador_Cassio.zip)

Regards,
Marcelo

**Reply by semiautomaticgit** [reply]
Hi @Maranhao6456 , 
it seems that the scpx file was created with a previous version of the software.
Unfortunately you can only try to import the vector ROIs extracted from the file, which I attach.
[geometry.gpkg.zip](https://github.com/semiautomaticgit/SemiAutomaticClassificationPlugin/files/14266225/geometry.gpkg.zip)

**Reply by Maranhao6456** [reply]
Hi Luca,

Unfortunately I was not successful in importing, even with your help. The screens follow.
![geometry](https://github.com/semiautomaticgit/SemiAutomaticClassificationPlugin/assets/67442712/885ef935-bca4-4bd1-a9ad-40db1d4d265e)
![qgis_scp](https://github.com/semiautomaticgit/SemiAutomaticClassificationPlugin/assets/67442712/7b55a24d-2eb8-4d3c-bc49-58e91aafd62c)

**Reply by Maranhao6456** [reply]
How did you convert my file from .scpx to . gpkg?

**Comment by semiautomaticgit** [comment]
> Hi Luca,
> 
> Unfortunately I was not successful in importing, even with your help. The screens follow. ![geometry](https://private-user-images.githubusercontent.com/67442712/304431857-885ef935-bca4-4bd1-a9ad-40db1d4d265e.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc4NDU3MjUsIm5iZiI6MTcwNzg0NTQyNSwicGF0aCI6Ii82NzQ0MjcxMi8zMDQ0MzE4NTctODg1ZWY5MzUtYmNhNC00YmQxLWE5YWQtNDBkYjFkNGQyNjVlLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjEzVDE3MzAyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZhODgyMGZmYjEyNjBlMDNkZWU1ZGZmNDY3MTQ1N2RhNzExOTk2MTk5N2ZmZWZiN2ExMDNlM2Q4YzAwZmRhYjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.kO670krycKbbZqT9j8vEtnRPiFr5f7LmJ4N7F_VxcBM) ![qgis_scp](https://private-user-images.githubusercontent.com/67442712/304431865-7b55a24d-2eb8-4d3c-bc49-58e91aafd62c.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc4NDU3MjUsIm5iZiI6MTcwNzg0NTQyNSwicGF0aCI6Ii82NzQ0MjcxMi8zMDQ0MzE4NjUtN2I1NWEyNGQtMmViOC00ZDNjLWJjNDktNThlOTFhYWZkNjJjLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjEzVDE3MzAyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFmNDQ4MTkxZjExNzkzM2YyOWU5YzY2Y2YzN2Y3ODVkZjNiN2VmZWQwNmMxZjMxY2FiMmE5YWFhMWRmN2JhMTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.PAvDTM6EhxEwj8QcqkVLG0uvZ8OFZVWKDdlW00_gL4k)

Hi @Maranhao6456 , it seems that you didn't create a training input file before importing


>How did you convert my file from .scpx to . gpkg?

I extracted the .scpx file which is actually a .zip file
